workspaces/global-floating-action-button/plugins/global-floating-action-button/src/components/FloatingButtons.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import { FloatingActionButton, Slot } from '../types';
+import { evaluateFloatingButtonsWithPositions } from '../utils';
+import { FloatingButton } from './FloatingButton';
+
+export const FloatingButtons = ({
+  floatingButtons,
+}: {
+  floatingButtons: FloatingActionButton[];
+}) => {
+  const floatingButtonMap =
+    evaluateFloatingButtonsWithPositions(floatingButtons);
+
+  return (
+    <>
+      {Object.keys(floatingButtonMap).map(fb => (
+        <FloatingButton
+          position={fb as Slot}
```

Comment:
Hmmmm, see issue is that Object.keys returns a list of strings, not a list of Slots here.

I'm a big fan of Record, but maybe it's worth to change the `FloatingActionButtonWithPositions` type here to an array so that it's easier to handle this in the React world:

```tsx
export type FloatingActionButtonWithPositions = Array<{
  slot: Slot,
  actionButtons: FloatingActionButton[],
}>;
```

I'm not sure because this makes the util more "complex" and you might also need to use the slot as key here.

Btw, the key is missed here anyway.